src/core/processing/qgsprocessingparameters.h

```diff
@@ -2841,6 +2856,40 @@ class CORE_EXPORT QgsProcessingParameterLayoutItem : public QgsProcessingParamet
     int mItemType = -1;
 };
 
+/**
+ * \class QgsProcessingParameterColor
+ * \ingroup core
+ * A color parameter for processing algorithms.
+  * \since QGIS 3.10
+ */
+class CORE_EXPORT QgsProcessingParameterColor : public QgsProcessingParameterDefinition
+{
+  public:
+
+    /**
+     * Constructor for QgsProcessingParameterColor.
+     */
+    QgsProcessingParameterColor( const QString &name, const QString &description = QString(), const QVariant &defaultValue = QVariant(),
```

Comment:
What do you think about adding a flag to control whether opacity is enabled for the parameter?

Reply:
Good idea, done